### ISSUE TITLE
Keepalived seg fault in docker

### ISSUE THREAD

**Issue by schmitch** [acassen/keepalived#758]
Keepalived Version: 1.4.0
Hello, currently I try to run keepalived in docker and getting the following segfault:
```
Mon Jan 22 10:43:47 2018: Starting Keepalived v1.4.0 (unknown)
Keepalived[7]: Starting Keepalived v1.4.0 (unknown)
Mon Jan 22 10:43:47 2018: Running on Linux 4.14.11-coreos #1 SMP Fri Jan 5 11:00:14 UTC 2018 (built for Linux 4.4.98)
Keepalived[7]: Running on Linux 4.14.11-coreos #1 SMP Fri Jan 5 11:00:14 UTC 2018 (built for Linux 4.4.98)
Mon Jan 22 10:43:47 2018: Opening file '/etc/keepalived/keepalived.conf'.
Keepalived[7]: Opening file '/etc/keepalived/keepalived.conf'.
Mon Jan 22 10:43:47 2018: Starting Healthcheck child process, pid=8
Keepalived[7]: Starting Healthcheck child process, pid=8
Mon Jan 22 10:43:47 2018: Starting VRRP child process, pid=9
Mon Jan 22 10:43:47 2018: Opening file '/etc/keepalived/keepalived.conf'.
Keepalived[7]: Starting VRRP child process, pid=9
Keepalived_healthcheckers[8]: Opening file '/etc/keepalived/keepalived.conf'.
Mon Jan 22 10:43:47 2018: Registering Kernel netlink reflector
Keepalived_vrrp[9]: Registering Kernel netlink reflector
Mon Jan 22 10:43:47 2018: Registering Kernel netlink command channel
Keepalived_vrrp[9]: Registering Kernel netlink command channel
Mon Jan 22 10:43:47 2018: Registering gratuitous ARP shared channel
Keepalived_vrrp[9]: Registering gratuitous ARP shared channel
Mon Jan 22 10:43:47 2018: Opening file '/etc/keepalived/keepalived.conf'.
Keepalived_vrrp[9]: Opening file '/etc/keepalived/keepalived.conf'.
Mon Jan 22 10:43:47 2018: Unknown keyword 'lvs_sync_daemon_inteface'
Keepalived_vrrp[9]: Unknown keyword 'lvs_sync_daemon_inteface'
Mon Jan 22 10:43:47 2018: ------< Global definitions >------
Keepalived_vrrp[9]: ------< Global definitions >------
Mon Jan 22 10:43:47 2018:  Router ID = master01.envisia.io
Keepalived_vrrp[9]:  Router ID = master01.envisia.io
Mon Jan 22 10:43:47 2018:  Default interface = eth0
Keepalived_vrrp[9]:  Default interface = eth0
Mon Jan 22 10:43:47 2018:  LVS flush = false
Keepalived_vrrp[9]:  LVS flush = false
Mon Jan 22 10:43:47 2018:  VRRP IPv4 mcast group = 224.0.0.18
Keepalived_vrrp[9]:  VRRP IPv4 mcast group = 224.0.0.18
Mon Jan 22 10:43:47 2018:  VRRP IPv6 mcast group = ff02::12
Keepalived_vrrp[9]:  VRRP IPv6 mcast group = ff02::12
Mon Jan 22 10:43:47 2018:  Gratuitous ARP delay = 5
Keepalived_vrrp[9]:  Gratuitous ARP delay = 5
Mon Jan 22 10:43:47 2018:  Gratuitous ARP repeat = 5
Keepalived_vrrp[9]:  Gratuitous ARP repeat = 5
Mon Jan 22 10:43:47 2018:  Gratuitous ARP refresh timer = 0
Keepalived_vrrp[9]:  Gratuitous ARP refresh timer = 0
Mon Jan 22 10:43:47 2018:  Gratuitous ARP refresh repeat = 1
Keepalived_vrrp[9]:  Gratuitous ARP refresh repeat = 1
Mon Jan 22 10:43:47 2018:  Gratuitous ARP lower priority delay = 5
Keepalived_vrrp[9]:  Gratuitous ARP lower priority delay = 5
Mon Jan 22 10:43:47 2018:  Gratuitous ARP lower priority repeat = 5
Keepalived_vrrp[9]:  Gratuitous ARP lower priority repeat = 5
Mon Jan 22 10:43:47 2018:  Send advert after receive lower priority advert = true
Keepalived_vrrp[9]:  Send advert after receive lower priority advert = true
Mon Jan 22 10:43:47 2018:  Send advert after receive higher priority advert = false
Keepalived_vrrp[9]:  Send advert after receive higher priority advert = false
Mon Jan 22 10:43:47 2018:  Gratuitous ARP interval = 0
Keepalived_vrrp[9]:  Gratuitous ARP interval = 0
Mon Jan 22 10:43:47 2018:  Gratuitous NA interval = 0
Keepalived_vrrp[9]:  Gratuitous NA interval = 0
Mon Jan 22 10:43:47 2018:  VRRP default protocol version = 3
Keepalived_vrrp[9]:  VRRP default protocol version = 3
Mon Jan 22 10:43:47 2018:  Iptables input chain = INPUT
Keepalived_vrrp[9]:  Iptables input chain = INPUT
Mon Jan 22 10:43:47 2018:  Using ipsets = false
Keepalived_vrrp[9]:  Using ipsets = false
Mon Jan 22 10:43:47 2018:  ipset IPv4 address set = keepalived
Keepalived_vrrp[9]:  ipset IPv4 address set = keepalived
Mon Jan 22 10:43:47 2018:  ipset IPv6 address set = keepalived6
Keepalived_vrrp[9]:  ipset IPv6 address set = keepalived6
Mon Jan 22 10:43:47 2018:  ipset IPv6 address,iface set = keepalived_if6
Keepalived_vrrp[9]:  ipset IPv6 address,iface set = keepalived_if6
Mon Jan 22 10:43:47 2018:  VRRP check unicast_src = false
Keepalived_vrrp[9]:  VRRP check unicast_src = false
Mon Jan 22 10:43:47 2018:  VRRP skip check advert addresses = false
Keepalived_vrrp[9]:  VRRP skip check advert addresses = false
Mon Jan 22 10:43:47 2018:  VRRP strict mode = false
Keepalived_vrrp[9]:  VRRP strict mode = false
Mon Jan 22 10:43:47 2018:  VRRP process priority = 0
Keepalived_vrrp[9]:  VRRP process priority = 0
Mon Jan 22 10:43:47 2018:  VRRP don't swap = false
Keepalived_vrrp[9]:  VRRP don't swap = false
Mon Jan 22 10:43:47 2018:  Checker process priority = 0
Keepalived_vrrp[9]:  Checker process priority = 0
Mon Jan 22 10:43:47 2018:  Checker don't swap = false
Keepalived_vrrp[9]:  Checker don't swap = false
Mon Jan 22 10:43:47 2018:  SNMP keepalived disabled
Keepalived_vrrp[9]:  SNMP keepalived disabled
Mon Jan 22 10:43:47 2018:  SNMP checker disabled
Keepalived_vrrp[9]:  SNMP checker disabled
Mon Jan 22 10:43:47 2018:  SNMP traps disabled
Keepalived_vrrp[9]:  SNMP traps disabled
Mon Jan 22 10:43:47 2018:  SNMP socket = default (unix:/var/agentx/master)
Keepalived_vrrp[9]:  SNMP socket = default (unix:/var/agentx/master)
Mon Jan 22 10:43:47 2018:  Network namespace = (default)
Keepalived_vrrp[9]:  Network namespace = (default)
Mon Jan 22 10:43:47 2018:  Script security disabled
Keepalived_vrrp[9]:  Script security disabled
Mon Jan 22 10:43:47 2018:  Default script uid:gid 0:0
Keepalived_vrrp[9]:  Default script uid:gid 0:0
Mon Jan 22 10:43:47 2018: ------< VRRP Topology >------
Keepalived_vrrp[9]: ------< VRRP Topology >------
Mon Jan 22 10:43:47 2018:  VRRP Instance = master-vips
Keepalived_vrrp[9]:  VRRP Instance = master-vips
Mon Jan 22 10:43:47 2018:    Using VRRPv3
Keepalived_vrrp[9]:    Using VRRPv3
Mon Jan 22 10:43:47 2018:    Want State = BACKUP
Keepalived_vrrp[9]:    Want State = BACKUP
Mon Jan 22 10:43:47 2018:    Running on device = ens192
Keepalived_vrrp[9]:    Running on device = ens192
Mon Jan 22 10:43:47 2018:    Skip checking advert IP addresses = no
Keepalived_vrrp[9]:    Skip checking advert IP addresses = no
Mon Jan 22 10:43:47 2018:    Enforcing strict VRRP compliance = no
Keepalived_vrrp[9]:    Enforcing strict VRRP compliance = no
Mon Jan 22 10:43:47 2018:    Using src_ip = 192.168.179.201
Keepalived_vrrp[9]:    Using src_ip = 192.168.179.201
Mon Jan 22 10:43:47 2018:    Gratuitous ARP delay = 5
Keepalived_vrrp[9]:    Gratuitous ARP delay = 5
Mon Jan 22 10:43:47 2018:    Gratuitous ARP repeat = 5
Keepalived_vrrp[9]:    Gratuitous ARP repeat = 5
Mon Jan 22 10:43:47 2018:    Gratuitous ARP refresh timer = 0
Keepalived_vrrp[9]:    Gratuitous ARP refresh timer = 0
Mon Jan 22 10:43:47 2018:    Gratuitous ARP refresh repeat = 1
Keepalived_vrrp[9]:    Gratuitous ARP refresh repeat = 1
Mon Jan 22 10:43:47 2018:    Gratuitous ARP lower priority delay = 5
Keepalived_vrrp[9]:    Gratuitous ARP lower priority delay = 5
Mon Jan 22 10:43:47 2018:    Gratuitous ARP lower priority repeat = 5
Keepalived_vrrp[9]:    Gratuitous ARP lower priority repeat = 5
Mon Jan 22 10:43:47 2018:    Send advert after receive lower priority advert = true
Keepalived_vrrp[9]:    Send advert after receive lower priority advert = true
Mon Jan 22 10:43:47 2018:    Send advert after receive higher priority advert = false
Keepalived_vrrp[9]:    Send advert after receive higher priority advert = false
Mon Jan 22 10:43:47 2018:    Virtual Router ID = 150
Keepalived_vrrp[9]:    Virtual Router ID = 150
Mon Jan 22 10:43:47 2018:    Priority = 201
Keepalived_vrrp[9]:    Priority = 201
Mon Jan 22 10:43:47 2018:    Advert interval = 1000 milli-sec
Keepalived_vrrp[9]:    Advert interval = 1000 milli-sec
Mon Jan 22 10:43:47 2018:    Accept enabled
Keepalived_vrrp[9]:    Accept enabled
Mon Jan 22 10:43:47 2018:    Preempt disabled
Keepalived_vrrp[9]:    Preempt disabled
Mon Jan 22 10:43:47 2018:    Promote_secondaries disabled
Keepalived_vrrp[9]:    Promote_secondaries disabled
Mon Jan 22 10:43:47 2018:    Tracked interfaces = 1
Keepalived_vrrp[9]:    Tracked interfaces = 1
Mon Jan 22 10:43:47 2018:      ens192 weight 0
Keepalived_vrrp[9]:      ens192 weight 0
Mon Jan 22 10:43:47 2018:    Virtual IP = 1
Keepalived_vrrp[9]:    Virtual IP = 1
Mon Jan 22 10:43:47 2018:      192.168.179.210/24 dev ens192 scope global
Keepalived_vrrp[9]:      192.168.179.210/24 dev ens192 scope global
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = lo
Keepalived_vrrp[9]:  Name = lo
Mon Jan 22 10:43:47 2018:  index = 1
Keepalived_vrrp[9]:  index = 1
Mon Jan 22 10:43:47 2018:  IPv4 address = 127.0.0.1
Keepalived_vrrp[9]:  IPv4 address = 127.0.0.1
Mon Jan 22 10:43:47 2018:  IPv6 address = ::
Keepalived_vrrp[9]:  IPv6 address = ::
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  is RUNNING
Keepalived_vrrp[9]:  is RUNNING
Mon Jan 22 10:43:47 2018:  MTU = 65536
Keepalived_vrrp[9]:  MTU = 65536
Mon Jan 22 10:43:47 2018:  HW Type = LOOPBACK
Keepalived_vrrp[9]:  HW Type = LOOPBACK
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = ens192
Keepalived_vrrp[9]:  Name = ens192
Mon Jan 22 10:43:47 2018:  index = 2
Keepalived_vrrp[9]:  index = 2
Mon Jan 22 10:43:47 2018:  IPv4 address = 192.168.179.201
Keepalived_vrrp[9]:  IPv4 address = 192.168.179.201
Mon Jan 22 10:43:47 2018:  IPv6 address = fe80::20c:29ff:fe24:99cb
Keepalived_vrrp[9]:  IPv6 address = fe80::20c:29ff:fe24:99cb
Mon Jan 22 10:43:47 2018:  MAC = 00:0c:29:24:99:cb
Keepalived_vrrp[9]:  MAC = 00:0c:29:24:99:cb
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  is RUNNING
Keepalived_vrrp[9]:  is RUNNING
Mon Jan 22 10:43:47 2018:  MTU = 1500
Keepalived_vrrp[9]:  MTU = 1500
Mon Jan 22 10:43:47 2018:  HW Type = ETHERNET
Keepalived_vrrp[9]:  HW Type = ETHERNET
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = docker0
Keepalived_vrrp[9]:  Name = docker0
Mon Jan 22 10:43:47 2018:  index = 3
Keepalived_vrrp[9]:  index = 3
Mon Jan 22 10:43:47 2018:  IPv4 address = 172.17.0.1
Keepalived_vrrp[9]:  IPv4 address = 172.17.0.1
Mon Jan 22 10:43:47 2018:  IPv6 address = ::
Keepalived_vrrp[9]:  IPv6 address = ::
Mon Jan 22 10:43:47 2018:  MAC = 02:42:7e:7e:32:17
Keepalived_vrrp[9]:  MAC = 02:42:7e:7e:32:17
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  MTU = 1500
Keepalived_vrrp[9]:  MTU = 1500
Mon Jan 22 10:43:47 2018:  HW Type = ETHERNET
Keepalived_vrrp[9]:  HW Type = ETHERNET
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = flannel.1
Keepalived_vrrp[9]:  Name = flannel.1
Mon Jan 22 10:43:47 2018:  index = 4
Keepalived_vrrp[9]:  index = 4
Mon Jan 22 10:43:47 2018:  IPv4 address = 10.244.0.0
Keepalived_vrrp[9]:  IPv4 address = 10.244.0.0
Mon Jan 22 10:43:47 2018:  IPv6 address = fe80::a0a5:afff:fee7:326b
Keepalived_vrrp[9]:  IPv6 address = fe80::a0a5:afff:fee7:326b
Mon Jan 22 10:43:47 2018:  MAC = a2:a5:af:e7:32:6b
Keepalived_vrrp[9]:  MAC = a2:a5:af:e7:32:6b
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  is RUNNING
Keepalived_vrrp[9]:  is RUNNING
Mon Jan 22 10:43:47 2018:  MTU = 1450
Keepalived_vrrp[9]:  MTU = 1450
Mon Jan 22 10:43:47 2018:  HW Type = ETHERNET
Keepalived_vrrp[9]:  HW Type = ETHERNET
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = cni0
Keepalived_vrrp[9]:  Name = cni0
Mon Jan 22 10:43:47 2018:  index = 5
Keepalived_vrrp[9]:  index = 5
Mon Jan 22 10:43:47 2018:  IPv4 address = 10.244.0.1
Keepalived_vrrp[9]:  IPv4 address = 10.244.0.1
Mon Jan 22 10:43:47 2018:  IPv6 address = fe80::2880:15ff:fece:661
Keepalived_vrrp[9]:  IPv6 address = fe80::2880:15ff:fece:661
Mon Jan 22 10:43:47 2018:  MAC = 0a:58:0a:f4:00:01
Keepalived_vrrp[9]:  MAC = 0a:58:0a:f4:00:01
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  is RUNNING
Keepalived_vrrp[9]:  is RUNNING
Mon Jan 22 10:43:47 2018:  MTU = 1450
Keepalived_vrrp[9]:  MTU = 1450
Mon Jan 22 10:43:47 2018:  HW Type = ETHERNET
Keepalived_vrrp[9]:  HW Type = ETHERNET
Mon Jan 22 10:43:47 2018: ------< NIC >------
Keepalived_vrrp[9]: ------< NIC >------
Mon Jan 22 10:43:47 2018:  Name = veth8dfe7399
Keepalived_vrrp[9]:  Name = veth8dfe7399
Mon Jan 22 10:43:47 2018:  index = 6
Keepalived_vrrp[9]:  index = 6
Mon Jan 22 10:43:47 2018:  IPv4 address = 0.0.0.0
Keepalived_vrrp[9]:  IPv4 address = 0.0.0.0
Mon Jan 22 10:43:47 2018:  IPv6 address = fe80::e81b:12ff:fe1b:cba3
Keepalived_vrrp[9]:  IPv6 address = fe80::e81b:12ff:fe1b:cba3
Mon Jan 22 10:43:47 2018:  MAC = ea:1b:12:1b:cb:a3
Keepalived_vrrp[9]:  MAC = ea:1b:12:1b:cb:a3
Mon Jan 22 10:43:47 2018:  is UP
Keepalived_vrrp[9]:  is UP
Mon Jan 22 10:43:47 2018:  is RUNNING
Keepalived_vrrp[9]:  is RUNNING
Mon Jan 22 10:43:47 2018:  MTU = 1450
Keepalived_vrrp[9]:  MTU = 1450
Mon Jan 22 10:43:47 2018:  HW Type = ETHERNET
Keepalived_vrrp[9]:  HW Type = ETHERNET
Mon Jan 22 10:43:47 2018: Using LinkWatch kernel netlink reflector...
Keepalived_vrrp[9]: Using LinkWatch kernel netlink reflector...
Mon Jan 22 10:43:47 2018: VRRP_Instance(master-vips) Entering BACKUP STATE
Keepalived_vrrp[9]: VRRP_Instance(master-vips) Entering BACKUP STATE
Mon Jan 22 10:43:48 2018: Keepalived_healthcheckers exited due to segmentation fault (SIGSEGV).
Keepalived[7]: Keepalived_healthcheckers exited due to segmentation fault (SIGSEGV).
Mon Jan 22 10:43:48 2018:   Please report a bug at https://github.com/acassen/keepalived/issues
Keepalived[7]:   Please report a bug at https://github.com/acassen/keepalived/issues
Mon Jan 22 10:43:48 2018:   and include this log from when keepalived started, what happened
Keepalived[7]:   and include this log from when keepalived started, what happened
Mon Jan 22 10:43:48 2018:   immediately before the crash, and your configuration file.
Keepalived[7]:   immediately before the crash, and your configuration file.
Mon Jan 22 10:43:48 2018: Healthcheck child process(8) died: Respawning
Keepalived[7]: Healthcheck child process(8) died: Respawning
Mon Jan 22 10:43:48 2018: Starting Healthcheck child process, pid=10
Keepalived[7]: Starting Healthcheck child process, pid=10
```

Config File is:
```
global_defs {
  vrrp_version 3
}

vrrp_instance master-vips {
  state BACKUP
  interface ens192
  lvs_sync_daemon_inteface ens192
  virtual_router_id 150
  priority 201
  nopreempt
  advert_int 1

  track_interface {
    ens192
  }

  virtual_ipaddress { 
    192.168.179.210/24
  }
}

# Service: kube-system/kube-haproxy-svc
virtual_server 192.168.179.210 6443 {
  delay_loop 5
  lvs_sched wlc
  lvs_method NAT
  persistence_timeout 1800
  protocol TCP

  
  real_server 192.168.179.201 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

  real_server 192.168.179.202 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

  real_server 192.168.179.203 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

}
```

I actually have no idea, why it is crashing.

Docker runs via:
` docker run --name keepalived -ti --cap-add=NET_ADMIN -v /etc/kubernetes/keepalived.conf:/etc/keepalived/keepalived.conf --net=host --security-opt label=disable --entrypoint bash keepalived:0.2 -s`

I can give build scripts for the image if needed.
actually ens192 is my host interface, and i'm in the 192.168.179.1-254 ip range. 
Currently I'm also having problems understanding the config. But as far as I know with `virtual_server` I can actually create a virtual server on top of the ip that load balances the real_servers like haproxy would do?

If I remove the `virtual_server` it actually starts working. any ideas?

**Comment by schmitch** [comment]
Guess it happens when the Port `6443` is already bound?

**Comment by pqarmitage** [comment]
Generally the reason for this is that the host system doesn't have the ip_vs module loaded, and it isn't possible to load the module from within the container.

Can you ensure that the host system has the ip_vs module loaded, and if not load it before starting the container.

**Comment by schmitch** [comment]
actually I loaded it manually via `modprobe ip_vs` also the module is loaded. because I run another instance of `keepalived` via kubernets on the same host (https://github.com/kubernetes/contrib/tree/master/keepalived-vip) this actually has a working `virtual_server` block, but uses a different version (1.2.24)

**Comment by pqarmitage** [comment]
I've just remembered that there is an issue with keepalived v1.4.0 when a virtual server doesn't have a sorry server configured. Commit 4955985 resolves this.

**Comment by schmitch** [comment]
@pqarmitage thanks, adding a sorry server fixed the issue ;)
OT: so if I have a virtual_server all reall_servers will be used like haproxy is that correct?

**Comment by pqarmitage** [comment]
Just an additional note:

1. `lvs_sync_daemon_inteface ens192` should have an `r` in `interface`
 
2. There is no benefit in specifying
```
  track_interface {
    ens192
  }
```
since that is the interface the vrrp instance is configured on, and so it will track it by default.

I'm afraid I'n not familiar with haproxy, so cannot comment on similarities to it.

**Comment by schmitch** [comment]
well just wanted to be sure that if I do curl 192.168.179.200:6443 that the IP is load balanced with 201,202 and 203

**Comment by pqarmitage** [comment]
The scheduling method you have is wlc, which the ipvsadm man page says:
```
wlc  -  Weighted  Least-Connection:  assigns  more  jobs to servers with fewer jobs and relative to the real servers' weight (Ci/Wi). This is the default.
```
So yes, there will be load balancing between 201, 202 and 203.

**Comment by schmitch** [comment]
thanks :)

**Comment by schmitch** [comment]
@pqarmitage 
I'm still having issues.
currently I can't access the service on 200:6443

actually this happens to be inside the logs:
```
IPVS (cmd 1160, errno 2): No such destination
```

**Comment by pqarmitage** [comment]
Can you please post your updated configuration and the full logs of keepalived (and please run it with the `-D` option) from when it starts up, so that we can see exactly what is happening. I will then attempt to reproduce the problem.

**Comment by schmitch** [comment]
after restarting with `-D` it started working. but I left everything else the same... strange.

**Comment by schmitch** [comment]
Nevermind after rebooting some instances I get the error again:
```
Jan 24 11:23:09 master03.envisia.io docker[20329]: Wed Jan 24 11:23:09 2018: VRRP_Instance(master-vips) Sending/queueing gratuitous ARPs on ens192 for 192.168.179.200
Jan 24 11:23:09 master03.envisia.io docker[20329]: Wed Jan 24 11:23:09 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:30:12 master03.envisia.io docker[20329]: Wed Jan 24 11:30:12 2018: TCP connection to [192.168.179.202]:tcp:6443 failed.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: TCP connection to [192.168.179.202]:tcp:6443 failed.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: Check on service [192.168.179.202]:tcp:6443 failed after 1 retry.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: Removing service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: IPVS (cmd 1160, errno 2): No such destination
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:32:02 master03.envisia.io docker[20329]: Wed Jan 24 11:32:02 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:32:02 master03.envisia.io docker[20329]: Wed Jan 24 11:32:02 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:32:02 master03.envisia.io docker[20329]: Wed Jan 24 11:32:02 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:32:22 master03.envisia.io docker[20329]: Wed Jan 24 11:32:22 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:32:22 master03.envisia.io docker[20329]: Wed Jan 24 11:32:22 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:32:22 master03.envisia.io docker[20329]: Wed Jan 24 11:32:22 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:32:42 master03.envisia.io docker[20329]: Wed Jan 24 11:32:42 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:32:42 master03.envisia.io docker[20329]: Wed Jan 24 11:32:42 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:32:42 master03.envisia.io docker[20329]: Wed Jan 24 11:32:42 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:33:02 master03.envisia.io docker[20329]: Wed Jan 24 11:33:02 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:33:02 master03.envisia.io docker[20329]: Wed Jan 24 11:33:02 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:33:02 master03.envisia.io docker[20329]: Wed Jan 24 11:33:02 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:33:22 master03.envisia.io docker[20329]: Wed Jan 24 11:33:22 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:33:22 master03.envisia.io docker[20329]: Wed Jan 24 11:33:22 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:33:22 master03.envisia.io docker[20329]: Wed Jan 24 11:33:22 2018: IPVS (cmd 1159, errno 2): No such file or directory
```

(output with `-D`)

Config:
```
global_defs {
  vrrp_version 3
}

vrrp_instance master-vips {
  state BACKUP
  interface ens192
  virtual_router_id 150
  priority 201
  nopreempt
  advert_int 1

  @master01 unicast_src_ip 192.168.179.201
  @master02 unicast_src_ip 192.168.179.202
  @master03 unicast_src_ip 192.168.179.203
  unicast_peer {
    @^master01 192.168.179.201
    @^master02 192.168.179.202
    @^master03 192.168.179.203
  }

  track_interface {
    ens192
  }

  virtual_ipaddress { 
    192.168.179.200
  }
}

virtual_server 192.168.179.200 6444 {
  delay_loop 20
  lvs_sched wlc
  lvs_method NAT
  persistence_timeout 1800
  protocol TCP
  ha_suspend

  sorry_server 192.168.179.201 6443

  real_server 192.168.179.201 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

  real_server 192.168.179.202 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

  real_server 192.168.179.203 6443 {
    weight 1
    TCP_CHECK {
      connect_port 6443
      connect_timeout 3
    }
  }

}
```

Docker Cmd:
`ExecStart=/usr/bin/docker run --name keepalived --cap-add=NET_ADMIN --privileged=true -v /lib/modules:/lib/modules -v /dev:/dev -v /etc/kubernetes/keepalived.conf:/etc/keepalived/keepalived.conf --net=host docker.envisia.io/envisia/keepalived:0.4 -D -i master03`

Docker Container:
```
FROM gcr.io/google_containers/ubuntu-slim:0.5

RUN apt-get update && apt-get install -y --no-install-recommends \
  libssl1.0.0 \
  libnl-3-200 \
  libnl-route-3-200 \
  libnl-genl-3-200 \
  iptables \
  libnfnetlink0 \
  libiptcdata0 \
  libipset3 \
  libipset-dev \
  libsnmp30 \
  kmod \
  ca-certificates \
  iproute2 \
  ipvsadm \
  bash && \
  apt-get clean && \
  rm -rf /var/lib/apt/lists/*

ADD keepalived.tar.gz /

RUN mkdir -p /etc/keepalived && \
  ln -s /keepalived/sbin/keepalived /usr/sbin && \
  ln -s /keepalived/bin/genhash /usr/sbin

COPY keepalived.conf /etc/keepalived/keepalived.conf

ENTRYPOINT ["/keepalived/sbin/keepalived","--dont-fork","--log-console", "-f","/etc/keepalived/keepalived.conf"]
```

Build with:
```
set -o pipefail

get_src()
{
  hash="$1"
  url="$2"
  f=$(basename "$url")

  curl -sSL "$url" -o "$f"
  echo "$hash  $f" | sha256sum -c - || exit 10
  tar xzf "$f"
  rm -rf "$f"
}

apt-get update && apt-get install -y --no-install-recommends \
  curl \
  gcc \
  libssl-dev \
  libnl-3-dev libnl-route-3-dev libnl-genl-3-dev iptables-dev libnfnetlink-dev libiptcdata0-dev \
  make \
  libipset-dev \
  git \
  libsnmp-dev \
  automake \
  ca-certificates

cd /tmp

# download, verify and extract the source files
get_src $SHA256 \
  "https://github.com/acassen/keepalived/archive/v$VERSION.tar.gz"

cd keepalived-$VERSION

aclocal
autoheader
automake --add-missing
autoreconf

./configure --prefix=/keepalived \
  --sysconfdir=/etc \
  --enable-snmp \
  --enable-sha1

make && make install

tar -czvf /keepalived.tar.gz /keepalived
```

**Comment by pqarmitage** [comment]
Could you please post the logs for keepalived from when it starts up, i.e. from and including the line that contains `Starting Keepalived`; there may be some messages there that will help diagnose what is happening.

**Comment by schmitch** [comment]
Sorry I wanted to add that, but I truncated at the wrong position.
```
Jan 24 11:21:22 master03.envisia.io docker[20308]: Error response from daemon: Cannot kill container: keepalived: Container 33130428311c8a43189ddefc95fbdf644d92f21955783a59a531357704707840 is not running
Jan 24 11:21:22 master03.envisia.io docker[20314]: keepalived
Jan 24 11:21:22 master03.envisia.io docker[20323]: 0.4: Pulling from envisia/keepalived
Jan 24 11:21:22 master03.envisia.io docker[20323]: Digest: sha256:ca0c33f9120f8518686f0b828e957fad23be0349a6b1a405c8a7a94f9b0195d7
Jan 24 11:21:22 master03.envisia.io docker[20323]: Status: Image is up to date for docker.envisia.io/envisia/keepalived:0.4
Jan 24 11:21:22 master03.envisia.io systemd[1]: Started keepalived.
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Starting Keepalived v1.4.0 (unknown)
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Running on Linux 4.14.11-coreos #1 SMP Fri Jan 5 11:00:14 UTC 2018 (built for Linux 4.4.98)
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Opening file '/etc/keepalived/keepalived.conf'.
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Starting Healthcheck child process, pid=5
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Opening file '/etc/keepalived/keepalived.conf'.
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Starting VRRP child process, pid=6
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Registering Kernel netlink reflector
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Initializing ipvs
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Registering Kernel netlink command channel
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Registering gratuitous ARP shared channel
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Registering Kernel netlink reflector
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Opening file '/etc/keepalived/keepalived.conf'.
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Gained quorum 1+0=1 <= 3 for VS [192.168.179.200]:tcp:6444
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Deactivating healthchecker for service [192.168.179.201]:tcp:6443 for VS [192.168.179.200]:tcp:6444
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: VRRP_Instance(master-vips) removing protocol VIPs.
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Deactivating healthchecker for service [192.168.179.202]:tcp:6443 for VS [192.168.179.200]:tcp:6444
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Deactivating healthchecker for service [192.168.179.203]:tcp:6443 for VS [192.168.179.200]:tcp:6444
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: Using LinkWatch kernel netlink reflector...
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: VRRP_Instance(master-vips) Entering BACKUP STATE
Jan 24 11:21:23 master03.envisia.io docker[20329]: Wed Jan 24 11:21:23 2018: VRRP sockpool: [ifindex(2), proto(112), unicast(1), fd(9,10)]
Jan 24 11:23:03 master03.envisia.io docker[20329]: Wed Jan 24 11:23:03 2018: VRRP_Instance(master-vips) Transition to MASTER STATE
Jan 24 11:23:03 master03.envisia.io docker[20329]: Wed Jan 24 11:23:03 2018: VRRP_Instance(master-vips) Received advert with lower priority 201, ours 201, forcing new election
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: VRRP_Instance(master-vips) Entering MASTER STATE
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: VRRP_Instance(master-vips) using locally configured advertisement interval (1000 milli-sec)
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: VRRP_Instance(master-vips) setting protocol VIPs.
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Netlink reflector reports IP 192.168.179.200 added
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: VRRP_Instance(master-vips) Sending/queueing gratuitous ARPs on ens192 for 192.168.179.200
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Activating healthchecker for service [192.168.179.200]:tcp:6444
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Activating healthchecker for service [192.168.179.200]:tcp:6444
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Activating healthchecker for service [192.168.179.200]:tcp:6444
Jan 24 11:23:04 master03.envisia.io docker[20329]: Wed Jan 24 11:23:04 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:08 master03.envisia.io docker[20329]: Wed Jan 24 11:23:08 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:08 master03.envisia.io docker[20329]: Wed Jan 24 11:23:08 2018: VRRP_Instance(master-vips) Sending/queueing gratuitous ARPs on ens192 for 192.168.179.200
Jan 24 11:23:08 master03.envisia.io docker[20329]: Wed Jan 24 11:23:08 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:08 master03.envisia.io docker[20329]: Wed Jan 24 11:23:08 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:08 master03.envisia.io docker[20329]: Wed Jan 24 11:23:08 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:09 master03.envisia.io docker[20329]: Wed Jan 24 11:23:09 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:23:09 master03.envisia.io docker[20329]: Wed Jan 24 11:23:09 2018: VRRP_Instance(master-vips) Sending/queueing gratuitous ARPs on ens192 for 192.168.179.200
Jan 24 11:23:09 master03.envisia.io docker[20329]: Wed Jan 24 11:23:09 2018: Sending gratuitous ARP on ens192 for 192.168.179.200
Jan 24 11:30:12 master03.envisia.io docker[20329]: Wed Jan 24 11:30:12 2018: TCP connection to [192.168.179.202]:tcp:6443 failed.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: TCP connection to [192.168.179.202]:tcp:6443 failed.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: Check on service [192.168.179.202]:tcp:6443 failed after 1 retry.
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: Removing service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:30:13 master03.envisia.io docker[20329]: Wed Jan 24 11:30:13 2018: IPVS (cmd 1160, errno 2): No such destination
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: Adding service [192.168.179.202]:tcp:6443 to VS [192.168.179.200]:tcp:6444
Jan 24 11:31:42 master03.envisia.io docker[20329]: Wed Jan 24 11:31:42 2018: IPVS (cmd 1159, errno 2): No such file or directory
Jan 24 11:32:02 master03.envisia.io docker[20329]: Wed Jan 24 11:32:02 2018: TCP connection to [192.168.179.202]:tcp:6443 success.
....
```

**Comment by pqarmitage** [comment]
I see in your config file that the sorry server is the same as one of the real servers. That will certainly cause some issues, so can you try with the sorry server being different from the real servers. Alternatively, apply patch 4955985 and then you won't need to configure a sorry server.

The logs from the startup of keepalived would still be helpful.

**Comment by pqarmitage** [comment]
After the `Activating healthchecker for service [192.168.179.200]:tcp:6444` but before the `TCP connection to [192.168.179.202]:tcp:6443 failed.` can you execute `ipvsadm -ln` (that's lower case L in the `-ln`), in the container and post the output. We should then be able to see what virtual and real servers are configured.

BTW, ipvs cmd 1160 is IP_VS_SO_SET_DELDEST and cmd 1159 is IP_VS_SO_SET_ADDDEST so it appears to be attempting to delete the failed real server and then add the real server.

One question that stands out is why is the connection to 192.168.179.202 alternately failing and then succeeding, whereas the connections to .201 and .203 appear to be successful.

It's nice to see the @... conditional configuration being used!

**Comment by schmitch** [comment]
Output is:
```
root@master03:/# ipvsadm -ln
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
```

@... is perfect for cloud-config ;)

**Comment by pqarmitage** [comment]
When you executed that command, was it executed in the docker container or in the host system. From my minimal knowledge of docker things, the command needed to be `docker exec ipvsadm -ln`

**Comment by schmitch** [comment]
I was inside the docker container `docker exec -ti ef92606cc8de bash` ;)

Basically I know created a new container from the latest commit (tried git apply to patch but somehow that failed). And will test this now.

**Comment by schmitch** [comment]
On the latest commit i get:
```
IP Virtual Server version 1.2.1 (size=4096)
Prot LocalAddress:Port Scheduler Flags
  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
TCP  192.168.179.200:6444 wlc persistent 1800
  -> 192.168.179.201:6443         Masq    1      0          0         
  -> 192.168.179.202:6443         Masq    1      0          0         
  -> 192.168.179.203:6443         Masq    1      2          0 
```
as an output, but I still have no connection.

**Comment by pqarmitage** [comment]
@schmitch Reluctantly I have come to the conclusion that the only way to resolve this is to build the container myself and work out what is happening, but I have very limited experience of using docker (hence my reluctance), and I will need some assistance.

I'm trying to build the docker container using the files you posted above, but unfortunately I am getting the following error (... replaces the list of my domain names for the x509 certificate):
```
docker build --pull -t build-keepalived:0.4 build
Sending build context to Docker daemon 3.584 kB
Step 1/6 : FROM gcr.io/google_containers/ubuntu-slim:0.5
Trying to pull repository gcr.io/google_containers/ubuntu-slim ... 
sha256:5e8ab08693fe72c57c4aab07192f11398095ab8c728bfdf9bd738e0c140ca3c1: Pulling from gcr.io/google_containers/ubuntu-slim
3e0b2de6a405: Downloading 
a3ed95caeb02: Download complete 
x509: certificate is valid for ..., not storage.googleapis.com
```
Do I need to install an x509 certificate somehow?

I'm also going to need to know how to amend the files so that I can build the docker container from a local tar file with keepalived sources.

**Comment by schmitch** [comment]
Hello, currently you do not need an x509 cert.

the files to build the container are here:
https://github.com/schmitch/keepalived-docker

you then can just run "make all". this will build keepalived in a container, put a binary tar.gz file in the root folder and copy it into a runnable container.

Basically if you want to run your config with keepalived you can just run it via `docker run -ti -v local_stored_keepalived.conf:/etc/keepalived/keepalived.conf envisia/keepalived`

**Comment by pqarmitage** [comment]
I've cloned your git repo and I'm getting exactly the same x509 error

**Comment by pqarmitage** [comment]
I've sorted out the problem; it seems it was something to do with a DNS query being blocked.

Could you please let me know how to make the necessary modifications so that I can build the docker image using a local tarball of the keepalived source code rather than downloading a released version from www.keepalived.org

**Comment by schmitch** [comment]
well I guess the problem is that the docker image is uploaded on storage.google.com (gcr.io) which could only be accessed if logged in via `gcloud` command.

I updated the repository.
Currently It uses the commit:
`497f1733dd95ae2ccf5e75ac09913da863113c66`

you can change the version by setting `build/Dockerfile` ENV VERSION to a commit and updating the sha256sum of the tarball (downloaded via https://github.com/acassen/keepalived/archive/$VERSION.tar.gz)

**Comment by pqarmitage** [comment]
I've now built a docker container as set out above and I have run your configuration with the docker image. Unfortunately I am not getting the errors that you are, although I am getting an error due to the sorry server being the same as one of the real servers.

You wrote earlier that you are running another instance of keepalived via kubernates in the same machine. I can certainly see that you have another instance of keepalived running since it logs `VRRP_Instance(master-vips) Received advert with lower priority 201, ours 201, forcing new election`.  Is that using the same configuration? If it is and it is running on the same host, even if it is in a container, that would be the cause of your problems since they would both be trying to add and manipulate the same ip_vs configuration.

There does appear to be a network connectivity issue, since master03 shouldn't become master if there is already a corresponding vrrp instance running with the same priority. Also why does the tcp check to 192.168.179.202 fail and then work again?

I have to admit that I am completely perplexed by the error message `IPVS (cmd 1159, errno 2): No such file or directory` (cmd 1159 is IP_VS_SO_SET_ADDDEST). I've looked at the kernel source code and so far as I can see, it cannot return ENOENT (errno 2) when the cmd is IP_VS_SO_SET_ADDDEST.

**Comment by schmitch** [comment]
actually this line is what I wondered aswell:
`VRRP_Instance(master-vips) Received advert with lower priority 201, ours 201, forcing new election`

however I do not have a second keepalived instance on the same host.
So it might be that the "correct" configuration is not used. or so. I try to investigate.
it might also be a problem that I use kubernetes+calico, etc..

**Comment by schmitch** [comment]
I'm actually closing the issue, since this problem is different from what I had anyway.
But still thanks for your patience.
